servers/server.go

```diff
@@ -601,6 +611,138 @@ func (p *Server) Extract(source, destination string) error {
 	return files.Extract(p.GetFileServer(), source, destination, "*", false, nil)
 }
 
+func (p *Server) CreateBackup() (string, int64, error) {
+	sourceFiles := []string{p.GetFileServer().Prefix()}
+
+	backupDirectory := p.RunningEnvironment.GetBackupDirectory()
+	if backupDirectory == "" {
+		return "", 0, pufferpanel.ErrSettingNotConfigured("backupDirectory")
+	}
+
+	if p.GetEnvironment().IsBackingUp() {
+		return "", 0, pufferpanel.ErrBackupInProgress
+	}
+
+	p.GetEnvironment().SetBackingUp(true)
+	defer p.GetEnvironment().SetBackingUp(false)
+
+	_, err := os.Stat(backupDirectory)
+	if err != nil && os.IsNotExist(err) {
+		err = os.Mkdir(backupDirectory, 0755)
+		if err != nil && !os.IsExist(err) {
+			return "", 0, err
+		}
+	}
+
+	backupId, err := uuid.NewV4()
+	if err != nil {
+		return "", 0, err
+	}
+	backupFileName := backupId.String() + ".tar.sz"
+	backupfile := path.Join(backupDirectory, backupFileName)
+
+	err = files.Compress(nil, backupfile, sourceFiles)
+	if err != nil {
+		return "", 0, err
+	}
+
+	file, err := os.Stat(backupfile)
+	if err != nil {
+		return "", 0, err
+	}
+
+	fileSize := file.Size()
+	return backupFileName, fileSize, err
+}
+
+func (p *Server) DeleteBackup(fileName string) error {
+	backupDirectory := p.RunningEnvironment.GetBackupDirectory()
+	if backupDirectory == "" {
+		return pufferpanel.ErrSettingNotConfigured("backupDirectory")
+	}
+
+	if p.GetEnvironment().IsBackingUp() {
+		return pufferpanel.ErrBackupInProgress
+	}
+
+	p.GetEnvironment().SetBackingUp(true)
+	defer p.GetEnvironment().SetBackingUp(false)
+
+	backupfile := path.Join(backupDirectory, fileName)
+
+	err := os.Remove(backupfile)
+	if err != nil && !os.IsNotExist(err) {
+		return err
+	}
+
+	return nil
+}
+
+func (p *Server) RestoreBackup(fileName string) error {
+	backupDirectory := p.RunningEnvironment.GetBackupDirectory()
+	if backupDirectory == "" {
+		return pufferpanel.ErrSettingNotConfigured("backupDirectory")
+	}
+
+	if p.GetEnvironment().IsBackingUp() {
+		return pufferpanel.ErrBackupInProgress
+	}
+
+	p.GetEnvironment().SetBackingUp(true)
+	defer p.GetEnvironment().SetBackingUp(false)
+
+	backupfile := path.Join(backupDirectory, fileName)
+
+	_, err := os.Stat(backupfile)
+	if err != nil && !os.IsNotExist(err) {
+		return err
+	}
+
+	serverfolder := p.GetFileServer().Prefix()
+
+	//Check if any files exist, as remove all errors if its empty
+	existingFiles, err := p.GetFileServer().Glob("*")
+	if err != nil {
+		return err
+	}
+
+	for _, existingFile := range existingFiles {
+		file, err := p.GetFileServer().Stat(existingFile)
+		if file.IsDir() {
+			err = p.GetFileServer().RemoveAll(existingFile)
+		} else {
+			p.GetFileServer().Remove(existingFile)
+		}
+		if err != nil {
+			logging.Info.Printf("failed to delted %s", err)
```

Comment:
typo for deleted

Reply:
fixed in [3afa64c](https://github.com/pufferpanel/pufferpanel/pull/1379/commits/3afa64cabbf2a66b294418fec744d28d5024bb52)